integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/app/ApiExceptionHandler.java

```diff
@@ -15,18 +15,24 @@
 import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 import uk.gov.crowncommercial.esourcing.integration.server.api.ApiUtil;
+import uk.gov.crowncommercial.esourcing.integration.service.EmailService;
+import uk.gov.crowncommercial.esourcing.integration.exception.SalesforceUpdateException;
 
 /*
  * Catches any errors raised by the API layer and returns a suitable error response.
  */
 @ControllerAdvice(basePackageClasses = ApiUtil.class)
-public class ApiExceptionHandler extends ResponseEntityExceptionHandler
- {
+public class ApiExceptionHandler extends ResponseEntityExceptionHandler {
+  private final EmailService emailService;
 
   private static final Logger LOG = LoggerFactory.getLogger(ApiExceptionHandler.class);
 
   @Autowired private Clock clock;
 
+  public ApiExceptionHandler(EmailService emailService) {
```

Comment:
Got a mixture of Autowired (the Clock) and constructor injection. One or the other, not both? I have no preference as to which.

Reply:
Hadn't noticed that, I prefer constructor.